host/src/main.rs

```diff
@@ -134,7 +134,10 @@ impl AsyncRead for Streams {
                                 return Poll::Ready(Ok(()));
                             }
                         }
-                        Err(StreamError::Closed) => return Poll::Ready(Ok(())),
+                        Err(StreamError::Closed) => {
+                            eprintln!("Closing stream");
```

Comment:
Let's leave the debug logging out for now.  If we get serious about debug logging, we can add a dependency on e.g. the `tracing` crate and go through and add `tracing::debug!` lines systematically, but I don't think we need that quite yet.